gen/src/generate/mod.rs

```diff
@@ -17,3 +17,11 @@ pub mod ucd;
 /// The standard auto-generated source disclaimer.
 const PREAMBLE: &'static str = "// WARNING: Auto-generated by the `unic-gen` crate.\n\
                                 // WARNING: DO NOT EDIT MANUALLY!";
+
+fn capitalize(str: &str) -> String {
```

Comment:
[std::ascii::AsciiExt::eq_ignore_ascii_case](https://doc.rust-lang.org/std/ascii/trait.AsciiExt.html#tymethod.eq_ignore_ascii_case)

Since property names are guaranteed to be ASCII only, this should work... (with some tweaks)

```rust
match str {
    $( stringify!($variant) => Ok($name::$variant), )*
    $( stringify!($abbr) => Ok($name::$abbr_variant), )*
    $( stringify!($long) => Ok($name::$long_variant), )*
    $( str if str.eq_ignore_ascii_case( stringify!($variant) ) => Ok($name::$variant), )*
    $( str if str.eq_ignore_ascii_case( stringify!($abbr) ) => Ok($name::$abbr_variant), )*
    $( str if str.eq_ignore_ascii_case( stringify!($long) ) => Ok($name::$long_variant), )*
    _ => Err(()),
}
```

Rust guarantees that when two match arms could both match, the first is taken, so it'll do the quicker matches check before the more expensive eq_ignore_ascii_case. I'll go try it out and send the PR.